#1786 Extend Application Insights Telemetry

OlegoO opened the pull request:
#1786 Extend Application Insights Telemetry with User Identity context information and X-Response-Time 

![image](https://user-images.githubusercontent.com/7639413/74411486-fd706b80-4e43-11ea-94a3-16dda9dc1852.png)

Commit:
#1786 Extend Application Insights Telemetry with User Identity context information and X-Response-Time

Commit:
Small refactoring

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/VirtoCommerce/vc-platform?pullRequest=1787) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/VirtoCommerce/vc-platform?pullRequest=1787) before we can accept your contribution.<br/><hr/>**OlegO** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/VirtoCommerce/vc-platform?pullRequest=1787) it.</sub>

yecli committed:
Merge branch 'dev' into feature/1786-extend-AI-telemetry

tatarincev commented on VirtoCommerce.Platform.Web/ApplicationInsights/CustomTelemetryInitializer.cs:
```diff
@@ -0,0 +1,31 @@
+            // Extend All telemetry with user identity context information.
+            if (httpContext != null && httpContext.User!=null)
+            {
+                 telemetry.Context.User.AuthenticatedUserId = httpContext.User.Identity.Name;
```
Could you simplify this block and make the code more safe from null-referenced exceptions?
```C#
telemetry.Context.User.AuthenticatedUserId = httpContext?.User?.Identity?.Name;
```

yecli replied:
If `AuthenticatedUserId` was filled before, we will erase it in case of `null` context, user or identity. Though do not know if it is possible to have `AuthenticatedUserId` filled and have e.g. `null` identity.

OlegoO replied:
@tatarincev  I could not find any requirements for AuthenticatedUserId. 
Only that Authenticated user id should be a persistent string that uniquely represents each authenticated user in the application or service.

Commit:
simplify this block and make the code more safe from null-referenced exceptions

Commit:
Merge branch 'feature/1786-extend-AI-telemetry' of https://github.com/VirtoCommerce/vc-platform into feature/1786-extend-AI-telemetry

yecli commented on VirtoCommerce.Platform.Web/App_Start/FilterConfig.cs:
```diff
@@ -21,26 +21,28 @@ public static void RegisterGlobalFilters(GlobalFilterCollection filters)
     /// </summary>
     public class ResponseTimeHeaderFilter : System.Web.Http.Filters.ActionFilterAttribute
```
Please address this comment

Commit:
Fix Make this class name end with 'Attribute'.
Fix  Startup.cs#L94: Make the enclosing instance method 'static' or remove this set on the 'static' field.
Fix FilterConfig.cs#L9: Add a 'protected' constructor or the 'static' keyword to the class declaration.
 Startup.cs#L719: Remove this hardcoded path-delimiter
